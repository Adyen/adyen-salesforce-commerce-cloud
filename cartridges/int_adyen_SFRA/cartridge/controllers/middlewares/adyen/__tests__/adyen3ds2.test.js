"use strict";

/* eslint-disable global-require */
var _require = require('../../index'),
    adyen3ds2 = _require.adyen.adyen3ds2;

var req;
var res;
var next;
beforeEach(function () {
  next = jest.fn();
  req = {
    https: true,
    host: 'mocked_host',
    querystring: {
      resultCode: 'Authorised',
      action: 'mocked_action',
      orderNo: 'mocked_orderNo'
    }
  };
  res = {
    redirect: jest.fn(),
    render: jest.fn()
  };
});
describe('Adyen 3DS2', function () {
  it('should render', function () {
    adyen3ds2(req, res, next);
    expect(res.render.mock.calls).toMatchSnapshot();
  });
  it('should throw', function () {
    var adyenGetOriginKey = require('*/cartridge/scripts/adyenGetOriginKey');

    var Logger = require('dw/system/Logger');

    var URLUtils = require('dw/web/URLUtils');

    adyenGetOriginKey.getOriginKeyFromRequest.mockImplementation(function () {
      throw Error('some_error');
    });
    adyen3ds2(req, res, next);
    expect(URLUtils.url.mock.calls).toMatchSnapshot();
    expect(Logger.error.mock.calls).toMatchSnapshot();
  });
});