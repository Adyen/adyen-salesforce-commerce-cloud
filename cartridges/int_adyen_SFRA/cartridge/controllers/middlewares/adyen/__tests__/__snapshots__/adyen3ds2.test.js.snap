// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adyen 3DS2 should render 1`] = `
Array [
  Array [
    "/threeds2/adyen3ds2",
    Object {
      "action": "mocked_adyen_action",
      "environment": "test",
      "locale": "nl_NL",
      "merchantReference": "mocked_orderNo",
      "originKey": "mocked_origin_key",
      "resultCode": "Authorised",
    },
  ],
]
`;

exports[`Adyen 3DS2 should throw 1`] = `
Array [
  Array [
    "Error-ErrorCode",
    "err",
    "general",
  ],
]
`;

exports[`Adyen 3DS2 should throw 2`] = `
Array [
  Array [
    "3DS2 redirect failed with reason: Error: some_error",
  ],
]
`;
